llvm/lib/CodeGen/AsmPrinter/DwarfFile.cpp

```diff
@@ -27,8 +29,20 @@ void DwarfFile::addUnit(std::unique_ptr<DwarfCompileUnit> U) {
 // Emit the various dwarf units to the unit section USection with
 // the abbreviations going into ASection.
 void DwarfFile::emitUnits(bool UseOffsets) {
-  for (const auto &TheU : CUs)
+  bool FirstContribution = true;
+  for (const auto &TheU : CUs) {
+    getOrCreateAbbrevSectionStart();
+    TheU.get()->setAbbrevSectionBegin(AbbrevSectionStart);
+    // If there is only one Abbreviation contribution or we are working with the
+    // first compile unit, set the Abbreviation Offset label to be the same as
+    // the section start label, do this so that when emitting the header, it
+    // emits the Abbreviation offset without doing any label math.
+    if (FirstContribution || Abbrevs.size() == 1) {
```

Comment:
I still don't understand why `|| Abbrevs.size() == 1` is necessary?

Reply:
Ah I see: It would be possible to have debug info that consists of multiple empty CUs that all share the same 1 abbreviation. Thanks!